tests/unit/test_charm.py

```diff
@@ -62,28 +58,29 @@ def test_credentials_are_propagated_only_username(self):
         # When only username is configured
         # Then it should toggle credentials configured to false
 
-        self.harness.update_config({"username": "a-username"})
-        self.assertFalse(self.harness.charm.credentials_configured)
-        self.assertIsInstance(self.harness.model.unit.status, BlockedStatus)
+        self.harness.update_config({"username": "a-username", "loki-url": "foo"})
```

Comment:
Feels like we could wrap these three calls in a helper method. Something like

```suggestion
        self._update_config_and_assert_status(config, expected_status)
```

Reply:
yeah but I was hesitant to spend too much time DRY'ing a bad test suite that I feel should be rewritten anyway (see #35)